tests/libvmifact/architecture.go

```diff
@@ -19,12 +19,69 @@
 
 package libvmifact
 
-var architecture string
+var (
+	architecture   string
+	minMemProvider MinimalMemoryProvider
+)
 
 func RegisterArchitecture(arch string) {
 	architecture = arch
+	if isARM64() {
```

Comment:
it may be prettier to have a map of providers indexed by the architecture.

Reply:
I don't see the point. This is not going to be change during runtime, so we are going to use one and only provider during the whole time the app (the test app) is running. There is no point to generate another provider we'll never use.

Reply:
My point is a design principle: the main flow should not be aware of how many providers are available. Adding a new provider should not involve adding another stanza in this if/else flow. One may even recommend that `RegisterArchitecture` would receive a single `Provider` argument, where the actual provider is defined outside of this file.

Reply:
ok. done.